Sources/MapboxDirections/VisualInstructionComponent.swift

```diff
@@ -124,11 +159,14 @@ public extension VisualInstruction.Component {
          - returns: A remote URL to the image.
          */
         public func imageURL(scale: CGFloat? = nil, format: Format = .png) -> URL? {
+            let imageScale = scale ?? ImageRepresentation.currentScale
             guard let imageBaseURL = imageBaseURL,
+                self.availableScales.contains(imageScale),
+                self.availableFormats.contains(format),
                 var imageURLComponents = URLComponents(url: imageBaseURL, resolvingAgainstBaseURL: false) else {
                 return nil
             }
-            imageURLComponents.path += "@\(Int(scale ?? ImageRepresentation.currentScale))x.\(format)"
+            imageURLComponents.path += "@\(Int(imageScale))x.\(format)"
```

Comment:
Does the Guidance Views API put `@1x` etc. and a file extension in the URL? If not, let’s use an empty `availableScales` or `availableFormats` set to detect that these things should not be added to the path.

Reply:
It doesn't its a direct URL that's the reason that I consider if you initialized an image with an imageURL you can use the property instead

Reply:
This line is unconditionally adding e.g. `@1x.png` to the URL. We should either avoid adding it to the URL or remove it from the URL before initializing the component. Let’s add a test to verify that the caller of this method will get the same URL that the Directions API returned for a guidance view image.